posts/2024-06-20-types-team-update.md

```diff
@@ -0,0 +1,225 @@
+actively maintained. Using the next-generation trait solver in rust-analyzer
+should result in a lot of additional testing for the solver and while also improving
+the IDE experience by positively impacting performance and correctness.
+
+We intend to slowly roll out the solver in other areas of the compiler until we're able
+to fully remove the existing implementation by the end of 2025. This switch will fix
+multiple unsound issues by itself and will unblock a significant amount of future work.
+It will generally cleanup many rough edges of the type system, such as associated types
+in binders. There are many unsound issues which can only be fixed once we exclusively
+use the new implementation. 
+
+### `a-mir-formality`
+
+* [`a-mir-formality` project goal](https://rust-lang.github.io/rust-project-goals/2024h2/a-mir-formality.html)
+
+We intend to more actively develop `a-mir-formality` this year to use it in our design process.
+Using it to model parts of the type system has already been incredibly impactful and we want
+to build on that. We are working on more effective testing of `a-mir-formality` by enabling its
+use for actual Rust code snippets and by adding fuzzing support. This will allow us to gain
+additional confidence in our model of the type system and will guide its future development.
+
+We plan to fully formalize some components of the type system this year. Coherence is fairly
+self-contained, very subtle, and soundness-critical. This has prevented us from making significant
+improvents to it in the past. We also intend to formalize coinductive trait semantics, which are
+difficult to reason about and necessary to fix many longstanding soundness issues.
+
+### Language changes and polonius
+
+* [Associated Type Position Impl Trait (ATPIT) project goal](https://rust-lang.github.io/rust-project-goals/2024h2/ATPIT.html)
+* [Polonius on Nightly project goal](https://rust-lang.github.io/rust-project-goals/2024h2/Polonius.html)
+
+We intend to get the internal implementation of opaque types ready for the stabilization
+of TAIT and ATPIT this year. We are also hoping to land a significant improvements our
+handling of associated types in coherence checking this year.
+
+Our other goal is to get [Polonius], the next generation borrow checker, available on nightly, which would put us in a position to stabilize in 2025 once we have time to do more optimization and testing.
+
+[polonius]: https://blog.rust-lang.org/inside-rust/2023/10/06/polonius-update.html
+
+We also intend to support the development of other language features, such as `async`-closures, which are part of the [async project goal](https://rust-lang.github.io/rust-project-goals/2024h2/async.html),
+and `dyn`-trait upcasting, which will hopefully get stabilized in the near future.
+
+## Roadmap
+
+### EOY 2024
+
+- next-generation trait solver
+    - stable in coherence
+    - used by rust-analyzer
+- TAIT/ATPIT stabilized
```

Comment:
```suggestion
- ATPIT stabilized
```

(Speaking as the language-side owner of this here:) It would surprise me (pleasantly, but still) if we stabilized TAIT in 2024.  It'd probably be better to not overpromise here and to just say that we're planning to stabilize ATPIT in 2024, which we are.